From 3278faed474a440bcd881d50a0146abdb0de8bc6 Mon Sep 17 00:00:00 2001
From: Anil Dsouza <anil.dsouza@einfochips.com>
Date: Mon, 14 Feb 2022 10:58:34 +0530
Subject: [PATCH 09/11] media: i2c: ap1320: Add support for arx3a0 sensor in
 ap1302 driver

* This patch adds support for ARX3A0 in AP1302. It supports 30 FPS
  with 560 x 560 resolution for this sensor.
* This patch also supports MEDIA_BUS_FMT_RGB565_2X8_LE format for
  AR0430 sensor with 10 FPS

Change-Id: I76dfd230e1d2835d9f288448047763d87775919e
Signed-off-by: Anil Dsouza <anil.dsouza@einfochips.com>
---
 drivers/media/i2c/ap1302.c | 26 ++++++++++++++++++++------
 1 file changed, 20 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index 0a5bac91e..72a275897 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -482,9 +482,9 @@ struct ap1302_firmware_header {
 
 static const struct ap1302_format_info supported_video_formats[] = {
 	{
-		.code = MEDIA_BUS_FMT_UYVY8_2X8,
-		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_YUV_JFIF
-			 | AP1302_PREVIEW_OUT_FMT_FST_YUV_422,
+		.code = MEDIA_BUS_FMT_RGB565_2X8_LE,
+		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_RGB
+			 | AP1302_PREVIEW_OUT_FMT_FST_RGB_565,
 	},
 };
 
@@ -535,8 +535,22 @@ static const struct ap1302_sensor_info ap1302_sensor_info[] = {
 		.name = "ar0430",
 		.i2c_addr = 0x36,
 		.resolution = { 2316, 1746},
-		.fps = 3,
-		.format = MEDIA_BUS_FMT_UYVY8_2X8,
+		.fps = 10,
+		.format = MEDIA_BUS_FMT_RGB565_2X8_LE,
+		.supplies = (const struct ap1302_sensor_supply[]) {
+			{ "vddpll", 0 },
+			{ "vaa", 0 },
+			{ "vdd", 0 },
+			{ "vddio", 0 },
+			{ NULL, 0 },
+		},
+	}, {
+		.model = "onnn,arx3a0",
+		.name = "arx3a0",
+		.i2c_addr = 0x36,
+		.resolution = { 560, 560},
+		.fps = 30,
+		.format = MEDIA_BUS_FMT_RGB565_2X8_LE,
 		.supplies = (const struct ap1302_sensor_supply[]) {
 			{ "vddpll", 0 },
 			{ "vaa", 0 },
@@ -2249,7 +2263,7 @@ static int ap1302_request_firmware(struct ap1302_device *ap1302)
 		return -EINVAL;
 	}
 
-	dev_dbg(ap1302->dev, "Requesting firmware %s\n", name);
+	dev_info(ap1302->dev, "Requesting firmware %s\n", name);
 
 	ret = request_firmware(&ap1302->fw, name, ap1302->dev);
 	if (ret) {
-- 
2.17.1

